File: new-resource-detector/src/test/java/nl/haploid/resource/detector/TestData.java
package nl.haploid.resource.detector;

import nl.haploid.event.RowChangeEvent;
import nl.haploid.resource.detector.service.ResourceDescriptor;

import java.util.Random;
import java.util.UUID;

public class TestData {

	public static long nextLong() {
		return new Random().nextLong();
	}

	public static ResourceDescriptor resourceDescriptor(final Long resourceId) {
		final ResourceDescriptor descriptor = new ResourceDescriptor();
		descriptor.setResourceId(resourceId);
		descriptor.setResourceType("olson");
		descriptor.setRowId(nextLong());
		descriptor.setTableName("draper");
		return descriptor;
	}

	public static String topic() {
		return UUID.randomUUID().toString();
	}

	public static RowChangeEvent rowChangeEvent(final String tableName) {
		final RowChangeEvent event = new RowChangeEvent();
		event.setId(nextLong());
		event.setRowId(nextLong());
		event.setTableName(tableName);
		return event;
	}
}

File: resource-builder/src/test/scala/nl/haploid/octowight/builder/TestData.scala
package nl.haploid.octowight.builder

import java.util.{Random, UUID}

import nl.haploid.octowight.registry.data.{Atom, ResourceIdentifier, ResourceMessage, ResourceRoot}

object TestData {

  def atom = new Atom(id = nextLong, origin = nextString, category = nextString)

  def nextLong = new Random().nextLong

  def nextString = UUID.randomUUID.toString

  def resource = new ResourceMessage(resourceIdentifier = resourceIdentifier, model = nextString, tombstone = false)

  def resourceIdentifier = new ResourceIdentifier(collection = nextString, id = nextLong)

  def resourceRoot =
    new ResourceRoot(resourceId = nextLong, resourceCollection = "captain", root = atom, version = nextLong)
}
